test/ModifyLiquidity.t.sol

```diff
@@ -0,0 +1,195 @@
+            IPoolManager.ModifyLiquidityParams({tickLower: -120, tickUpper: 120, liquidityDelta: 2e18, salt: SALT});
+
+        // Get the uninitialized positions and assert they have no liquidity.
+        Position.Info memory positionSalt = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+
+        Position.Info memory positionSalt2 = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter2), LIQ_PARAM_SALT_2.tickLower, LIQ_PARAM_SALT_2.tickUpper, SALT
+        );
+
+        assertEq(positionSalt.liquidity, 0);
+        assertEq(positionSalt2.liquidity, 0);
+
+        // Modify the liquidity with the salt with the first router.
+        modifyLiquidityRouter.modifyLiquidity(simpleKey, LIQ_PARAM_SALT, ZERO_BYTES);
+
+        Position.Info memory updatedPositionSalt = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+
+        Position.Info memory updatedPositionSalt2 = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter2), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+
+        // Assert only the liquidity from the first router is updated.
+        assertEq(updatedPositionSalt.liquidity, uint128(uint256(LIQ_PARAM_SALT.liquidityDelta)));
+        assertEq(updatedPositionSalt2.liquidity, 0);
+
+        // Modify the liquidity with the second router.
+        modifyLiquidityRouter2.modifyLiquidity(simpleKey, LIQ_PARAM_SALT_2, ZERO_BYTES);
+
+        updatedPositionSalt2 = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter2), LIQ_PARAM_SALT_2.tickLower, LIQ_PARAM_SALT_2.tickUpper, SALT
+        );
+        updatedPositionSalt = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_SALT.tickLower, LIQ_PARAM_SALT.tickUpper, SALT
+        );
+
+        // Assert only the liquidity from the second router is updated.
+        assertEq(updatedPositionSalt2.liquidity, uint128(uint256(LIQ_PARAM_SALT_2.liquidityDelta)));
+        assertEq(updatedPositionSalt.liquidity, uint128(uint256(LIQ_PARAM_SALT.liquidityDelta)));
+    }
+
+    function test_gas_modifyLiquidity_newPosition() public {
+        // Create rand position to warm the router. No gas snaps.
+        modifyLiquidityRouter.modifyLiquidity(simpleKey, LIQ_PARAM_NO_SALT, ZERO_BYTES);
+
+        modifyLiquidityRouter.modifyLiquidity(simpleKey, LIQ_PARAM_SALT, ZERO_BYTES);
+        snapLastCall("create new liquidity to a position with salt");
+
+        IPoolManager.ModifyLiquidityParams memory samePositionNewSalt = LIQ_PARAM_SALT;
+        samePositionNewSalt.salt = hex"BEEF";
+
+        modifyLiquidityRouter.modifyLiquidity(simpleKey, samePositionNewSalt, ZERO_BYTES);
+        snapLastCall("create new liquidity to same position, different salt");
```

Comment:
idk if both these gas snapshots are needed? Theyre identical - theyre both snapshots of adding a new position with a salt